docs/sources/writing-guide/shortcodes/index.md

```diff
@@ -143,6 +143,43 @@ This shortcode inserts a lists of links to the page's subpages and includes the
 {{</* section withDescriptions="true"*/>}}
 ```
 
+## `docs/reference` shortcode
+
+The `docs/reference` shortcode offers more flexible linking than the Hugo builtin `relref` shortcode.
+Reference links wrapped in the `docs/reference` shortcode can be defined with multiple destinations.
+The destination used in a rendered page derives from matching path prefixes:
+
+```markdown
+{{%/* docs/reference */%}}
+[<LABEL>]: "<PROJECT PATH PREFIX> -> <REFERENCE>"
+...
+{{%/* /docs/reference */%}}
+```
+
+- _`LABEL`_ matches a link label used in a reference-style link (`[LINK TEXT][LABEL])`).
+  For more information about reference-style links, refer to [Daring Fireball: Markdown Syntax Documentation](https://daringfireball.net/projects/markdown/syntax#link).
+- _`PROJECT PATH PREFIX`_ is matched against the page’s URL path, where the first matching `PROJECT PATH PREFIX` is chosen.
+- _`REFERENCE`_ is the parameter that would typically be used in a `relref` shortcode.
+  It can include the syntax `<SOMETHING VERSION>`, which is first looked up in the front matter before falling back to the version inferred from the page's version.
+
+### Example
+
+```markdown
+{{%/* docs/reference */%}}
+[grafana]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>"
+[grafana]: "/docs/grafana-cloud/ -> /docs/grafana-cloud"
+[grafana]: "/docs/mimir/ -> /docs/grafana/<GRAFANA VERSION>"
+{{%/* /docs/reference */%}}
+```
```

Comment:
I definitely need to rewrite this section because I think the behavior is currently easy to misundersand.

Although your understanding is close, the explanation in your comment does not match the behavior of the example usage.

Assuming the reference-style link:

```markdown
For more information about Grafana dashboards, refer to [Dashboards][dashboards].
```

The shortcode that would match your explanation would be:

```
{{% docs/reference %}}
[dashboards]: "/docs/grafana/ -> /docs/grafana/latest/dashboards"
[dashboards]: "/docs/grafana-cloud/ -> /docs/grafana/latest/dashboards"
[dashboards]: "/docs/mimir/ -> /docs/grafana/latest/dashboards"
{{% /docs/reference %}}
```

Since there is no contextual destination, we don't even need this shortcode for the preceding example, and can just use:

```markdown
For more information about Grafana dashboards, refer to [Dashboards]({{< relref "/docs/grafana/latest/dashboards" >}}).
```

One problem with linking the in the preceding example is that we don't usually want to link to `latest` but instead to the current version of documentation.
We don't want `/docs/grafana/next/` content linking to `/docs/grafana/latest/`.

That's where the version variable `<GRAFANA VERSION>` comes in.

Now the same shortcode but with version interpolation:

```markdown
{{% docs/reference %}}
[dashboards]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/dashboards"
[dashboards]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/dashboards"
[dashboards]: "/docs/mimir/ -> /docs/grafana/<GRAFANA VERSION>/dashboards"
{{% /docs/reference %}}
```

Now the front matter `grafana_version` will affect the reference.
If the front matter `grafana_version` is not present, then the version is inferred from the page's version.

1. If the shortcode is rendered in the page `/docs/grafana/latest/alerting/`, **without the** front matter `grafana_version`, the reference used would become `/docs/grafana/latest/dashboards`.
1. If the shortcode is rendered in the page `/docs/grafana-cloud/alerting/`, **with** the front matter `grafana_version: next`, the reference used would become
`/docs/grafana/next/dashboards`.


Does this explain things more clearly?

Reply:
My example isn't relevant as anything besides a guide for the kind of example I'm looking for to ensure I can set one of these up. Having a for-example would just be helpful to me if I understand the scenario and the intended outcome and it uses the version variable. I think the part of your explanation where you actually show how you use the label in the context of some text is **essential** for understanding how to use these.

I've added some comments to try to clarify what I think the meanings are for each part of the setup.  But I'm still not clear on the what I need to enter for the GRAFANA VERSION variable? Nothing? Am I just letting the the system sort that out unless I have a reason to point to some specific version?

Reply:
Yes the `<SOMETHING VERSION>` variable syntax is interpolated by the shortcode. 

It is currently documented in https://github.com/grafana/writers-toolkit/pull/264/files#diff-bdc57a00bf7e2ae51ef4333951207adc6f3b3fd4d36ffe7389a5222f1591152cR171

and updated verbiage is proposed in https://github.com/grafana/writers-toolkit/pull/264#discussion_r1230807343

Reply:
> My example isn't relevant as anything besides a guide for the kind of example I'm looking for to ensure I can set one of these up.

My bad, I assumed because of the similar content that your example was a reinterpretation of the existing example.

Reply:
I've tried writing more concrete examples in https://github.com/grafana/writers-toolkit/pull/264/commits/1f92907c3f0198f74045d74c4cd845db0f9b93d2